solr/core/src/java/org/apache/solr/cloud/api/collections/BackupCmd.java

```diff
@@ -138,6 +138,15 @@ public void call(ClusterState state, ZkNodeProps message, NamedList<Object> resu
 
       backupMgr.writeBackupProperties(backupProperties);
 
+      // It can't be done within aggregateResults call
+      // since endTime is filled later
+      if(backupProperties != null) {
```

Comment:
Within `aggregateResults` the backup properties are nullable but here it (currently) seems always present so perhaps the null-check here could be omitted?

Reply:
Hi @cpoerschke Thanks for the note! It can be `null`, in case of non incremental call it will be redirected to `copyIndexFiles(URI, String, ZkNodeProps, NamedList<Object>)`:

```java
  public void call(ClusterState state, ZkNodeProps message, NamedList<Object> results) throws Exception {
    ...
    try (BackupRepository repository = cc.newBackupRepository(repo)) {
      ...
      if (incremental) {
          incrementalCopyIndexFiles(backupUri, collectionName, message, results, backupProperties, backupMgr);
      } else {
          copyIndexFiles(backupUri, collectionName, message, results);
      }
      ...
    }
    ...
  }
```

and within `copyIndexFiles` it will call `aggregateResults` without providing `backupManager` and `backupProps`:

```java
  private void copyIndexFiles(URI backupPath, String collectionName, ZkNodeProps request, NamedList<Object> results) throws Exception {
    ...
    NamedList<Object> aggRsp = aggregateResults(results, collectionName, slices, null, null);
    ...
  }
```

Maybe we should investigate this and it is safe to provide `copyIndexFiles` with `backupManager` and `backupProps`?

Reply:
Ah, sorry, I misunderstood you. Yeah, sure, at that point it is always non null. Will remove the check...